status/k8s.go

```diff
@@ -479,10 +479,42 @@ func (k *K8sStatusCollector) status(ctx context.Context) *Status {
 
 	err := k.podStatus(ctx, status, defaults.AgentDaemonSetName, "k8s-app=cilium", func(ctx context.Context, status *Status, name string, pod *corev1.Pod) {
 		if pod.Status.Phase == corev1.PodRunning {
+			// extract container status
+			var containerStatus *corev1.ContainerStatus
+			for i, cStatus := range pod.Status.ContainerStatuses {
+				if cStatus.Name == defaults.AgentContainerName {
+					containerStatus = &pod.Status.ContainerStatuses[i]
+					break
+				}
+			}
 			tasks = append(tasks, statusTask{
 				name: pod.Name,
 				task: func(_ context.Context) error {
-					s, err := k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
+					var s *models.StatusResponse
+					var err error
+
+					if containerStatus != nil && containerStatus.State.Running != nil {
+						// if container is running, execute "cilium status" in the container and parse the result
+
+						s, err = k.client.CiliumStatus(ctx, k.params.Namespace, pod.Name)
+
+					} else {
+						// otherwise, generate a useful status message
+						desc := "is not running"
+
+						// determine CrashLoopBackOff status. Yes, it is this ugly.
+						if containerStatus != nil {
+							if containerStatus.State.Waiting != nil && containerStatus.State.Waiting.Reason == "CrashLoopBackOff" {
+								desc = "is in CrashLoopBackOff"
+							}
+							if containerStatus.LastTerminationState.Terminated != nil {
+								desc = fmt.Sprintf("%s, exited with code %d", desc, containerStatus.LastTerminationState.Terminated.ExitCode)
+							}
+						}
+						err = fmt.Errorf("container %s %s, try 'kubectl -n %s logs -c %s %s'", defaults.AgentContainerName, desc,
```

Comment:
Can we just do this for the user? Even if cilium-cli just does this for the first agent that has the problem, it would be an improvement. I wouldn't want it to do this N times for N nodes in the cluster.

Background: In Cilium CI like #1026, the cluster was already destroyed so I can't really run the command.

Reply:
Sure, that's a good ideal